private/bufpkg/bufplugin/bufplugin.go

```diff
@@ -90,6 +93,28 @@ func PluginToProtoPluginRegistryType(plugin Plugin) registryv1alpha1.PluginRegis
 	return registryType
 }
 
+// LanguagesToProtoLanguages determines the appropriate registryv1alpha1.PluginRegistryType for the plugin.
+func LanguagesToProtoLanguages(languages []string) []registryv1alpha1.PluginLanguage {
+	languageToEnum := make(map[string]registryv1alpha1.PluginLanguage)
+	for pluginLanguageKey, pluginLanguage := range registryv1alpha1.PluginLanguage_value {
+		pluginLanguageKey := strings.TrimPrefix(pluginLanguageKey, "PLUGIN_LANGUAGE_")
+		pluginLanguageKey = strings.ToLower(pluginLanguageKey)
+		// Example:
+		// { go: 1, javascript: 2 }
+		languageToEnum[pluginLanguageKey] = registryv1alpha1.PluginLanguage(pluginLanguage)
+	}
+	protoLanguages := []registryv1alpha1.PluginLanguage{}
+	for _, language := range languages {
+		if pluginLanguage, ok := languageToEnum[language]; ok {
```

Comment:
What should we do here if we don't find a language? Should this show up as an error to indicate we were only able to map some of them to the enum type?

Reply:
Ye, you're probably right. We'll want to error here, so if a user (us) refer to an unknown language it'll fail right away.

And then we'll have to file an issue in this repository to add that language.

Reply:
Add a useful error message here:

```
invalid plugin output language: "cobol"
supported languages: cpp, csharp, dart, go, java, javascript, kotlin, objective_c, php, python, ruby, rust, swift, typescript
```